### PR TITLE
The Waku Network: Technical Overview and Properties

### PR DESCRIPTION
**Pull request by fryorcraken**

### CODE REVIEW

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+Thanks to the enhancements made to discv5, nodes are able to know what shards other peers are servicing before connecting to them.
+
+### Message Routing - Autosharding
+
+One issue with sharding is that users and applications need to know what shard to use.
+
+One application could arbitrarily decide what shard they want to use but that may not scale easily, especially when adding new shards. It is also one more decision for the developer to make; we prefer to make the dev ex as easy as possible. It also gives the opportunity for developers to build applications that spread on several shards without delegating shard choices to the user.
+
+Autosharding is a simple protocol that dispatches messages on a shard based on the application.
```
Comment (jm-clius) [inline posts/2024-02-15-waku-network-tech-overview.mdx:112]: ...by mapping the application automatically to an underlying routing shard.

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+### Message Routing - RLN Relay
+
+Waku is generalized, meaning any kind of payload can be transported. Hence, there is no strict definition of “spam”. Whether the message contains a [meme](https://github.com/b4s36t4/waku-meme) or a [zk note for private DeFi](https://railgun.ch/#/), Waku shouldn't even know the content; metadata and data should remain private.
```
Comment (jm-clius) [inline posts/2024-02-15-waku-network-tech-overview.mdx:116]: "Waku is generalized _and privacy-preserving_"
We do define spam (as high rate messages). There is no strict definition of "spam content".

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+So rather than defining what a "spam" message looks like, we introduced rate limit on the network to enable a fair use of Waku, with a capping of the bandwidth usage per shard.
+
+This is done with RLN, or Rate Limiting Nullifier, which enables to limit the rate of messages sent by a given publisher.
+It is currently set to 1 msg/s. Coupled with a maximum message size (150kB) and a maximum number of publishers (80k, TBD), we are able to assume a maximum bandwidth usage per shard (around 10Mbps).
+
+Rate-limiting publishers in a censorship-resistant and private manner is difficult; this is why we are using zero-knowledge technology:
+
+1. Users push their RLN credentials to a smart contract (currently on Ethereum Sepolia testnet).
+2. Nodes watch the contract and include the new membership.
```
Comment (jm-clius) [inline posts/2024-02-15-waku-network-tech-overview.mdx:133]: Nodes keep track of all registered memberships on the contract.

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+3. When sending a message, the user attaches an RLN proof to the message with the current epoch (timestamp in seconds).
+4. Nodes can verify the proof, without being able to correlate the user’s Ethereum address (used on the smart contract) with the message being sent, preserving anonymity.
+5. If a user attempts to send more than 1 msg/s, nodes can detect this and drop the message surplus, or spam, protecting the network.
```
Comment (jm-clius) [inline posts/2024-02-15-waku-network-tech-overview.mdx:136]: Importantly: and disconnect from peers forwarding spam, thereby isolating the spam origin from the network.

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+### Servicing Mostly-offline and Resource-restricted Devices
+
+Finally, how is Waku useful to mobile and browsers?
```
Comment (jm-clius) [inline posts/2024-02-15-waku-network-tech-overview.mdx:140]: resource-restricted devices, such as mobile phones and web browsers?

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+Waku defines a number of [request-reply protocols](https://rfc.vac.dev/spec/10/#requestreply-domain) to enable such devices to access the Waku network, without having to be always online or consume extensive amounts of bandwidth, i.e., without participating in the Waku Relay network.
```
Comment (jm-clius) [inline posts/2024-02-15-waku-network-tech-overview.mdx:142]: *request-response (trying to bring some consistency to our terminology across platforms)

File: posts/2024-02-15-waku-network-tech-overview.mdx (hunk 5 above)
Comment (fryorcraken) [reply posts/2024-02-15-waku-network-tech-overview.mdx:140]: "resource-restricted devices, " is such a mouthful I am trying to find a way to alternate a bit to make it easier to read. Thoughts @Amelia7689 ?

File: posts/2024-02-15-waku-network-tech-overview.mdx (hunk 6 above)
Comment (fryorcraken) [reply posts/2024-02-15-waku-network-tech-overview.mdx:142]: I also prefer "request-response" but our specs says "request-reply": https://rfc.vac.dev/spec/10/#requestreply-domain
Very happy to deprecate "request-reply" I think it sounds weird.

Comment (jm-clius) [reply posts/2024-02-15-waku-network-tech-overview.mdx]: Yeah, I don't think we've been very consistent, but I'd rather we update RFC 10 to "request-response" (most other RFCs already use "request-response", e.g. Store here: https://rfc.vac.dev/spec/13/#wire-specification)

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+- State of things
+- USPs
+- Work moving forward
+-->
+
+Last December [we announced The Waku Network](https://thedefiant.io/waku-launches-first-decentralised-privacy-preserving-dos-protections-for-peer-to-peer-messaging).
+
+In the press release, we defined the Waku Network as the first decentralized and privacy-preserving network that provides Denial-of-Service (DoS) protections for peer-to-peer messaging. Aiming to enhance privacy and security in peer-to-peer communication by implementing innovative protocols and technologies.
+
+Let’s dig a bit deeper on the why and what is The Waku Network.
```
Comment (Amelia7689) [inline posts/2024-02-15-waku-network-tech-overview.mdx:30]: 'Let us'

We should stick with not using contractions throughout the article

Comment (Amelia7689) [reply posts/2024-02-15-waku-network-tech-overview.mdx:30]: We should also standardise capitalisation of 'the Waku Network'

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+
+## Origin
+
+If you have been following Waku or Status, you are likely to be familiar with the origin of Waku.
+The Status mobile app was created to be a Web3 superapp, a portal to the Ethereum ecosystem that leverages the three original pillars: Ethereum for consensus, Swarm for storage and Whisper for communication.
```
Comment (Amelia7689) [inline posts/2024-02-15-waku-network-tech-overview.mdx:35]: 'super app'

File: posts/2024-02-15-waku-network-tech-overview.mdx [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+
+The Status development team attempted to build Status with Whisper, but there were fundamental limitations to this protocol, especially for resource-restricted devices, aka, mobile phones.
```
Comment (Amelia7689) [inline posts/2024-02-15-waku-network-tech-overview.mdx:37]: 'but this protocol had fundamental limitations'

File: posts/2024-02-15-waku-network-tech-overview.mdx (hunk 7 above)
Comment (fryorcraken) [reply posts/2024-02-15-waku-network-tech-overview.mdx:30]: @Amelia7689 are you able to take over and proceed with corrections? feel free to open a 2nd PR and I can close this one.

Here is a link to get the file: https://github.com/waku-org/blog.waku.org/blob/waku-network-tech/posts/2024-02-15-waku-network-tech-overview.mdx

**Comment by fryorcraken** [comment]
https://github.com/waku-org/blog.waku.org/pull/27

**Pull request closed without merging**